src/media/inner.rs

```diff
@@ -1322,10 +1321,12 @@ impl MediaInner {
         self.simulcast = Some(s);
     }
 
-    pub fn ssrc_rx_for_rid(&self, repairs: Rid) -> Option<Ssrc> {
+    pub fn find_primary_ssrc_for_rid(&self, repairs: Option<Rid>, rtx_ssrc: Ssrc) -> Option<Ssrc> {
+        // Find the most recent matching ssrc, in case the ssrc has changed. We ignore sources with
+        // repairs, as those are RTX, and we ignore ourselves.
         self.sources_rx
             .iter()
-            .find(|r| r.rid() == Some(repairs))
+            .rfind(|r| r.rid() == repairs && r.ssrc() != rtx_ssrc && !r.is_rtx())
```

Comment:
Do we still need to ignore ourselves if we, ourselves, have .is_rtx() == true?

Reply:
yes. cause if this is a new source.. we don't have "repairs" set yet, so is_rtx will return false.